src/include/mpir_handlemem.h

```diff
@@ -57,15 +57,25 @@ int MPIR_check_handles_on_finalize(void *objmem_ptr);
 
 /* This routine is called by finalize when MPI exits.  This routine may be called
  * before initialization and after finalization of MPI. */
-static inline int MPIR_Handle_free(void **indirect, int indirect_size)
+static inline int MPIR_Handle_free(void **indirect, int indirect_size, bool no_mpl)
```

Comment:
I am okay to create `MPIR_Handle_free_impl()` or `MPIRI_Handle_free()`, but you also rejected the idea of `_impl()` in the first round of review. I will keep this, but please let me know any better idea so that I can remove this extra parameter.

Reply:
The current `MPIR_Handle_free()` takes `MPIR_Object_alloc_t`, so it can check its handle type (i.e., whether it is info or not) in MPIR_Handle_free()` without an extra parameter. No new function for this.